notebooks/maze_tuto.ipynb

```diff
@@ -0,0 +1,642 @@
+    ")\n",
+    "# instanciate the domain\n",
+    "domain = domain_factory()\n",
+    "# init the start position\n",
+    "domain.reset()\n",
+    "# display the corresponding maze\n",
+    "display(domain.render())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Solvers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2021-07-19T09:23:28.408241Z",
+     "start_time": "2021-07-19T09:23:28.404886Z"
+    }
+   },
+   "source": [
+    "### Finding suitable solvers\n",
+    "The library hub includes a lot of solvers. We can use `match_solvers` function to show available solvers that fit the characteristics of the defined domain, according to the mixin classes used to define the [domain type](#domain-type). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "match_solvers(domain=domain)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the following, we will restrict ourself to 2 solvers:\n",
+    "\n",
+    "- `StableBaseline`, quite generic, allowing to use reinforcement learning (RL) algorithms by wrapping a stable OpenAI Baselines solver ([stable_baselines3](https://github.com/DLR-RM/stable-baselines3))\n",
+    "- `LazyAstar` (A*), more specific, coming from path planning."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### PPO solver"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We apply the PPO solver to the maze problem and rollout quickly the proposed solution to see if it was able to actually solve the maze. We set a maximum number of steps to reach the goal according to maze size.\n",
+    "\n",
+    "We use the `with ...` pattern so that `solver.cleanup()` is automatically called after solving the problem.\n",
+    "Note that this implies to play also the solution inside the `with ...` statement."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "with StableBaseline(\n",
+    "    PPO, \"MlpPolicy\", learn_config={\"total_timesteps\": 300}, verbose=True\n",
+    ") as solver:\n",
+    "    # solve\n",
+    "    print(\"** Solving ...\", flush=True)\n",
+    "    MazeDomain.solve_with(solver, domain_factory)\n",
+    "    # rollout\n",
+    "    domain = domain_factory()\n",
+    "    max_steps = maze.width * maze.height\n",
+    "    print(f\"** Playing solution  with max_steps={max_steps} ...\", flush=True)\n",
+    "    rollout(\n",
+    "        domain,\n",
+    "        solver,\n",
+    "        num_episodes=1,\n",
+    "        max_steps=max_steps,\n",
+    "        max_framerate=None,\n",
+    "        outcome_formatter=None,\n",
+    "        render=False,\n",
```

Comment:
I did change this to True ... (assuming a user would try to do so) but the graphic is not displayed.

Reply:
I think it would be best to solve and then write our own roll out ...

Reply:
I did that, (writting the rollout) but in A* (following the narratives). I can put it here if necessary. 
The bad part is that we need to do that at the same time as the solve because of this "with ..." syntax that scikit-decide want to enforce when using solvers.

Reply:
The basic usage is to call from from inside a context manager; it is okay IMO to emulate it without a context manager, but the important point is to call `solver.close()` at the end.